### ISSUE TITLE
Using the model type "function" does not converge

### ISSUE THREAD

**Issue by LuisFMCuriel** [feilab-hust/VCD-Net#1]
Hi,

First of all, congratulations on this amazing work. The way to use it is very intuitive and it provides an incredibly fast way to reconstruct Light Field images.

I have been using the UnetA for the reconstruction of different structures in C. elegans and Zebrafish. However, when I train using the "function" type model, the training never converges. The ln_loss keeps giving me the same value for each epoch (independently of the dataset that I use). I achieved good reconstructions using the structure type model, but when I train with the other Unet (UnetB), the loss never changes over the training (it always gives a value of ~0.965). The same thing happens when I enable edge_loss. I have tried different datasets for this but it doesn't seem to work for this model (the datasets work well for the structure model - UnetA).

I checked the code and both unets use the same loss function, so maybe is a defect in the UnetB? I would really appreciate if you can give me some feedback.

Thanks in advance,
Luis.

**Comment by CQ-Yi** [comment]
Hi,
'function' tpye model is lighter than 'structure' one because we reduce the number of layers and remove BN layer for quantitative accuracy. So when meeting some complicated data (like zebrafish), it's hard to make training converge.

**Comment by LuisFMCuriel** [comment]
Hi,
Yes, I saw that it's lighter. We are trying to reconstruct neurons in C. elegans (same as [here](https://www.nature.com/articles/s41592-021-01058-x?proof=t#citeas)) but the training doesn't converge independently the dataset we use (the loss function doesn't even vary). What do you suggest us to change to make it work?
